### ISSUE TITLE
Check all crypto plugins before release

### ISSUE THREAD

**Issue by grthor** [jcryptool/crypto#287]
Before Release 1.0, the correct function of all plug-ins should be checked.

In the following list all plugins are listed. After you have checked one plugin, you can check it off here, so all others will see which plugins are already processed and there will be no double checks.

Here is a little check list that you can use:
- Buttons in the plugin toolbar working (top right corner of the plugin (restart button etc.))
- F1 shows context help
- Is an online help available (theory and instruction manual)
- Play one time throught the plugin: All Buttons working, no major GUI problems (e.g. text it cut off)
- Repeat the workflow in the plugin in a different order and compare results. 
- Check for Typos
- Change size: Are scrollbars shown, if not all can be displayed within the window? 
- Are buttons and groupings (frames) consistent in names (text) and position (layout)? 
- If a new dialog pops up, is the focus for keyboard entry set to the first entry field, so a user can just continue typing? 
- All items of this list should be checked in German and Englisch 

## Analysis
Just enter your name in the appropriate column if you have checked or fixed a plugin.

Plugin | geprüft von | überarbeitet von | Kommentar
---|---|---|---
Entropy Analysis | @grthor | @grthor | Kein Titel und Beschreibung, aber sonst gut. Ich würde es so durchgehen lassen. Das Plugin ist in sich schlüssig.
Frequency Analysis | @grthor | @grthor  | ~~(Y-Achse des Graphen verschiebt sich nach oben beim verschieben des Graphen ("Mehr Optionen" auswählen, "Referenzverteilung" aktivieren, Referenztext ändern, Graph verschieben dann tritt das Problem auf)~~ fixed with https://github.com/jcryptool/crypto/commit/9ad835c31529a3a9f121a2d997e3e149a6413f93
Friedman Test | @grthor | @grthor | 
Grille Analysis | @grthor | @grthor |
Substitution Analysis | @grthor | @grthor |
Transposition Analysis | @grthor | @grthor |
Vignère Breaker | @grthor | @grthor | ~~Tooltips on the 2nd and 3rd page have random positions~~ fixed with https://github.com/jcryptool/crypto/commit/8a6d35bcb1f5060305811cc2088e01060935e7b6 ~~y-Achse verschwindet bei der Häufigkeitsanalyse (vgl. Frequency Analysis Plugin Problem)~~ fixed with https://github.com/jcryptool/crypto/commit/9ad835c31529a3a9f121a2d997e3e149a6413f93)
Viterbi Analysis | @grthor | @grthor |

## Visuals
Just enter your name in the appropriate column if you have checked or fixed a plugin.

Plugin | geprüft von | überarbeitet von | Kommentar
---|---|---|---
Android Unlock Pattern | @grthor | @grthor |
Ant Colony Optimization | @grthor | @grthor | ~~Letters in the visualization overlap. They are not readable~~ -- fixed
ARC4 / Spritz | @grthor | @grthor |
Certificate Verification | @grthor | @grthor | Changes from https://github.com/jcryptool/crypto/issues/261 included
Chinese Remainder Theorem | @grthor | @grthor |
Diffie-Helman Key Exchange | @grthor | @grthor |
[ElGamal Cryptosystem](https://github.com/jcryptool/crypto/issues?q=is%3Aissue+is%3Aopen+ElGamal) |          |           | fix before 1.0 
Elliptic Curve Calculations | @be4 | @grthor | 
Extended Euclidean / Reciprocal Subtraction | @grthor | @grthor |
Extended RSA Cryptosystem | @grthor | @grthor |
Feige Fiat Shamir | @grthor | @grthor |
Fiat Shamir | @grthor | @grthor |
Graph Isomorphism | @grthor | @grthor |
[Grille](https://github.com/jcryptool/crypto/issues/287#issuecomment-611590471) | @simlei, @grthor  | @grthor  |UTF8, ~resize~, ~minor stuff~
[Hash Sensitivity](https://github.com/jcryptool/crypto/issues/287#issuecomment-611636271) | @simlei, @grthor  | @grthor | scrolling, ~context help~ 
[Homomorphic encryption](https://github.com/jcryptool/crypto/issues/287#issuecomment-613280317) | @simlei, @grthor  |    | one possible functional bug, rest only minor issues
[Huffman Coding](https://github.com/jcryptool/crypto/issues/287#issuecomment-610998143) | @simlei, @grthor  |     | ~Context help~, **Malfunction** (UTF8)
[Inner States of the Data Encryption Standard](https://github.com/jcryptool/crypto/issues/287#issuecomment-611025830) | @simlei | @simlei | ~Context help~, intuitiveness, but overall OK
[Kleptography](https://github.com/jcryptool/crypto/issues/287#issuecomment-611042258) | @simlei | @simlei |  ~minor context help issue~, rest ok
[Magic Door](https://github.com/jcryptool/crypto/issues/287#issuecomment-613267099) | @simlei, @grthor | @grthor |  pretty good, ~~minor context help problems~~ -- @grthor context help problems fixed
[Merkle Signature (XMSS^MT)](https://github.com/jcryptool/crypto/issues/287#issuecomment-611064801) | @simlei) | @simlei, @grthor  |  pretty good, awesome graph! :open_mouth: 
[Merkle-Hemlman Knapsack Cryptosystem](https://github.com/jcryptool/crypto/issues/287#issuecomment-611558552) | @simlei | @simlei, @grthor |
Multipartite Key Exchange |   @grthor       |       |  No context and online help
[RSA Cryptosystem](https://github.com/jcryptool/crypto/issues/287#issuecomment-613265678) | @simlei |        | **needs thorough fixing and re-testing**
[Shamirs Secret Sharing](https://github.com/jcryptool/crypto/issues/287#issuecomment-604983186) (ctd. https://github.com/jcryptool/crypto/issues/291) | @simlei | @simlei | Prio:  ?? **medium/high** + low ??
[Shanks Babystep-Giantstep](https://github.com/jcryptool/crypto/issues/287#issuecomment-605041294) | @simlei | @simlei | Prio: ?? **medium (~context help~)** + low ??
[Signature Demonstration](https://github.com/jcryptool/crypto/issues/287#issuecomment-605072738) | @simlei | @simlei | Prio: ?? **medium** (~context help~, orphaned shells) + low ??
[Signature Verification](https://github.com/jcryptool/crypto/issues/287#issuecomment-605107380) | @simlei | @simlei | Prio: ?? **medium**: orphaned shells, ~context help~, low (focus) ??
[Simple Power Analysis / Square and Multiply](https://github.com/jcryptool/crypto/issues/287#issuecomment-605456083) | @simlei |  @grthor     | ~Prio: **medium** for bad scaling in a visual; low/medium(?): textual issues~ fixed with https://github.com/jcryptool/crypto/commit/b745c4162a65dc24c4d06d840e6894dc80b7bfb7 by @grthor 
[SPHINCS Signature](https://github.com/jcryptool/crypto/issues/287#issuecomment-605294624) | @simlei |        |  Prio: **low/medium** ~context help~ 
[SPHINCS+ Signature](https://github.com/jcryptool/crypto/issues/287#issuecomment-605438817) | @simlei  |        | Prio: **medium/high** visualization + ~context help~
[SSl / TLS Handshake](https://github.com/jcryptool/crypto/issues/287#issuecomment-605443955) | @simlei |          | Prio: **medium/high** bug/glitch**, low/medium textual/~context help~ 
[Verifiable Secret Sharing](https://github.com/jcryptool/crypto/issues/287#issuecomment-605446873) | @simlei | @simlei | no issues at all found :tada:
[WOTS / WOTS+](https://github.com/jcryptool/crypto/issues/287#issuecomment-605451129) | @simlei |           | Prio: **medium** for badly-scaled visual

## Games
Just enter your name in the appropriate column if you have checked or fixed a plugin.

Plugin | geprüft von | überarbeitet von | Kommentar
---|---|---|---
[Divider Game](https://github.com/jcryptool/crypto/issues/287#issuecomment-605492896) | @tassadarius | @tassadarius | **fixed**
[Number Shark](https://github.com/jcryptool/crypto/issues/287#issuecomment-605500954) | @tassadarius |     | Prio: **medium** fix the optimizer output 
~~[Sudoku Solver](https://github.com/jcryptool/crypto/issues/287#issuecomment-609025500)~~ | ~~@tassadarius~~ |   | ~~Prio: **medium/high** fix the StackOverflow error~~ Plugin kommt nicht ins Release 1.0!
Zudo-Ku |   |     |

**Comment by simlei** [comment]
### El Gamal:

Resolution: *reasonably resolve all problems before v1.0*

**Open issues:**

- https://github.com/jcryptool/crypto/issues/288
- https://github.com/jcryptool/crypto/issues/218
- https://github.com/jcryptool/crypto/issues/223

**Misc. observations:**

**Comment by simlei** [comment]
## Shamirs Secret Sharing

continued here: https://github.com/jcryptool/crypto/issues/291

### Preliminary verdict:

- Low-prio: Online help in English not ideal
- **Mid/High-prio**: visualization doesnt work (tested under Linux, other OS tests?) s. I1), visualization not smooth (I2)

### English

### Both languages

- Possibly just Linux: visual on the right does not work (System: Linux, GTK3), s. I1)
- Under Windows (see screenshot below) there is a graph, but it is
wrong: As shown in the online help it's a smooth graph, but here it has
edges. And the checked shares should be on the graph and not besides it. Can be it was once working so maybe checking the history of the plugin could help. S. I2)

1) [x] Online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [x] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
2) [x] Walk-through major functions
    - all fine except the visual on the right (System: Linux, GTK3)
    - see I1)
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
    - no other than that in 2)
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Textual:**

- Online Help:
    - _"Here is shown the numerical mode."_
    - _"By pressing the Select button opens a dialog box "_
    - _"Press the OK button the dialog box is closed and the polynomial represented as a formula."_
    - _"The user has now the opportunity There to generate a prime "_
    - ... more
    - Online Help is nicely formatted except for the images which are always fixed-width. Better would be percentage of page width.
    - Online Help in English is not ideal as for sentence construction, grammar.

### German

1) [x] online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [x] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
2) [x] walk-through major functions
    - same as in English
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
    - same as in English
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**I1)**

![Shamir1](https://user-images.githubusercontent.com/982379/77757667-ef605e00-7031-11ea-9f29-2d5768a43498.gif)

**I2)**

![dplnknhcoapdceja](https://user-images.githubusercontent.com/982379/77779769-4dea0400-7053-11ea-9425-f99c69ec6fdf.png)

**Comment by Leikani** [comment]
I can look at at the online help.

> # Shamirs Secret Sharing
> Possibly just Linux: WOTS+ visualization on the right does not work
What does Shamir's Secret Sahring have to do with WOTS+?

Best regards, BE


Am 27.03.2020 um 13:51 schrieb Simon Leischnig:
> # Shamirs Secret Sharing
>
> ### English
>     
> - Possibly just Linux: WOTS+ visualization on the right does not work (System: Linux, GTK3)
>
> [x] 1) Online help (theory, instruction)
>     - [x] has context help
>     - [x] has online help
>     - [x] has toolbar button that shows context help
>     - [o] do online help images scale with page width?
> [x] 2) Walk-through major functions
>     - all fine except the WOTS+ visualization on the right (System: Linux, GTK3)
>     - see I1)
> [x] 3) Walk-through, different order
> [x] 4) Typos
> [x] 5) Resize artifacts
>     - no other than that in 2)
> [x] 6) Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
> [x] 7) Dialogs: autofocus on first field?
>
> **Textual:**
>
> - Online Help:
>     - _"Here is shown the numerical mode."_
>     - _"By pressing the Select button opens a dialog box "_
>     - _"Press the OK button the dialog box is closed and the polynomial represented as a formula."_
>     - _"The user has now the opportunity There to generate a prime "_
>     - ... more
>     - Online Help is nicely formatted except for the images which are always fixed-width. Better would be percentage of page width.
>     - Online Help in English is not ideal as for sentence construction, grammar.
>
> ### German
>
> **Textual:**
>
> [x] 1) online help (theory, instruction)
>     - [x] has context help
>     - [x] has online help
>     - [x] has toolbar button that shows context help
>     - [o] do online help images scale with page width?
> [x] 2) walk-through major functions
>     - same problem as in English
> [x] 3) walk-through, different order
> [x] 4) Typos
> [x] 5) Resize artifacts
> [x] 6) Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
> [x] 7) Dialogs: autofocus on first field?
>
> **I1)**
>
> ![Shamir1](https://user-images.githubusercontent.com/982379/77757667-ef605e00-7031-11ea-9f29-2d5768a43498.gif)
>
>

**Comment by simlei** [comment]
## Shanks Babystep-Giantstep

### Preliminary Verdict:

 - ~*medium prio* for missing context help~ fixed

- Titel- und Beschreibung nicht per Standard, unnötige Scrollbar. Wird bald flächendeckend gelöst. s. I1)

### English

1) [x] Online help (theory, instruction)
    - [ ] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Textual:**

(to be fixed in a wider-scope effort)
- Online Help:
    - "_Shanks Babystep-Giantstep plug-in could be started by the menu Visualization_" --> "_The Shanks Babystep-Giantstep plug-in can be started by the menu Visualization_"
    - _"The explanations adjust corresponding to the current step"_ --> "The explanations reflect the current step"
    - "_The view of plug-in_" --> "_The view of the plugin_"
    - "_Then you can start the algorithm again with new user input**s**_" --> "_Then you can start the algorithm again with new user input_"
    - more...


### German

1) [x] online help (theory, instruction)
    - [ ] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

## Signature Demonstration

- Dialog opens useless shell that hangs around: s. I1)
    - same in hash function button, signature function button

**I1)**

![babystep](https://user-images.githubusercontent.com/982379/77768256-7fa69f00-7042-11ea-89c0-6479fef116e5.gif)

**Comment by simlei** [comment]
@Leikani 

> I can look at at the online help.

Thanks!

> What does Shamir's Secret Sahring have to do with WOTS+?

as shown in the referenced .gif, on the right there are WOTS+-labeled radiobuttons, and on their right there is a visualization area that doesn't paint.
**Edit** disregard that, these radio buttons are part of the described graphics artifact with the hanging visual. It was framebuffered over from another view.

**Comment by simlei** [comment]
## Signature Demonstration

### Preliminary verdict:

- ~medium to high priority: context help, useless blank shells~ (fixed)
- low-priority: rest

### Both

- Dialog opens useless shell that hangs around (annoying, looks bad): s. I1)
    - same in hash function button, signature function button
    - Under Linux/GTK3

### English

1) [x] Online help (theory, instruction)
2) [x] Walk-through major functions
    - [x] has context help
        - but the context help is "Demonstration of RSA"! --> fix!
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
3) [x] Walk-through, different order
4) [ ] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Textual:**

- Online Help:
    - Greeting formula as in German; remove
    - same link to the github page of the keystore as in German; remove

### German

1) [x] online help (theory, instruction)
    - [ ] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Textual:**

- Online Help:
    - Useless "greeting formula": remove. "_Willkommen in der Online Hilfe des Plug-Ins zur Signaturvisualisierung. Nachfolgenden finden Sie einen Überblick des Plug-Ins sowie eine detaillierte Erklärung zu jeden Schritt des Signaturerstellungsprozesses._"
        - also in English
    - Link: "_Besuchen Sie den Standard-JCT-Keystore_" -- a) does not work for me (no window opens); b) shows the github page for org.jcryptool.keystore. Not useful; remove.
        - also in English

**I1)**

![signaturedemo1](https://user-images.githubusercontent.com/982379/77774021-b4b6ef80-704a-11ea-9a85-89ab98aeb697.gif)

**Comment by be4** [comment]
In your gif, the graph with the WOTS shouldn't be there. Maybe a pointer
reference problem.
There should be a polynom graph.

Under Windows (see screenshot below) I see the polynom graph, but it is
wrong: As shown in the online help it's a smooth graph, but here it has
edges. And the checked shares should be on the graph and not besides it.

Here a correction seemed to have destroyed something.






There seems to be Am 27.03.2020 um 15:53 schrieb Simon Leischnig:
> @Leikani 
>
>> I can look at at the online help.
> Thanks!
>
>> What does Shamir's Secret Sahring have to do with WOTS+?
> as shown in the referenced .gif, on the right there are WOTS+-labeled radiobuttons, and on their right there is a visualization area that doesn't paint
>

**Comment by simlei** [comment]
> Under Windows (see screenshot below) I see the polynom graph

OK so probably a Linux/GTK specific bug. I'd say this is something to be fixed until v1.0 even if not all OS are affected.

> [...] but it is wrong: As shown in the online help it's a smooth graph, but here it has edges. And the checked shares should be on the graph and not besides it. Here a correction seemed to have destroyed something.

This should be another issue; I am taking this to the original comment. Please feel free to attach a screenshot there, too.

**Comment by simlei** [comment]
## Signature Verification

### Preliminary verdict:

- ~mid prio: orphaned shells~ (fixed)
- ~mid prio: context help~ (fixed)
- ~low prio: autofocus in dialog~ (fileselect button gets focus)

### Both languages

- Dialogs open useless shell that hangs around: s. I1)
    - same in hash function button, signature function button

### English

1) [x] Online help (theory, instruction)
    - [x] has context help
        - but for RSA! fix!
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?
    - [ ] "choose input" dialog is not autofocused on button or textfield

**Textual:**
    - Second tab label is a bit too wordy I think -- "_Validity according to the chain model_"-- simply "Validity/Gültigkeit" would suffice
    - same in German

### German

1) [x] online help (theory, instruction)
    - [ ] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?
    - [ ] "Input auswählen" dialog is not autofocused on button or textfield


**I1)**

![signatureverification](https://user-images.githubusercontent.com/982379/77780065-b76a1280-7053-11ea-9494-cd28c9f18cde.gif)

**Comment by simlei** [comment]
## SPHINCS Signature

### Preliminary verdict

- up until the context help integration, quite good!
- Prio **low/medium**: Context help integration

### English

1) [x] Online help (theory, instruction)
    - [ ] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### German

1) [x] online help (theory, instruction)
    - [ ] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Comment by simlei** [comment]
## SPHINCS+ Signature

### Preliminary verdict:

- **Prio Medium/high**: visualizations are broken under Linux and would confuse the user.
- **Medium**: context help

### Both languages

- The visualizations on linux are broken, s. I1). Sphincs without the plus seems to not have these problems with the same kind of visualization.
- text field below description in I2) has no content, I am not sure what it does.
- High CPU load: https://github.com/jcryptool/crypto/issues/258

### English

**Textual:**

- Online Help:
    - below the first "remark" there is a solitary dot. remove.

1) [x] Online help (theory, instruction)
    - [ ] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### German

1) [x] online help (theory, instruction)
    - [ ] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**I1)**

![sphincs+](https://user-images.githubusercontent.com/982379/77822681-4a0ebe00-70f5-11ea-842c-5adf3f259747.gif)

**I2)**

![sphincs+_empty](https://user-images.githubusercontent.com/982379/77822680-4418dd00-70f5-11ea-9104-c82bb8055762.png)

**Comment by simlei** [comment]
## SSl / TLS Handshake

### Preliminary verdict

- **Prio Medium/high** erroneus behavior without feedback (I1)
- **Prio low/medium** textual (translation, help), context help integration

### Both languages

- Not sure: is it necessary to disable "information" buttons when proceeding to next step?
  Information could be nice to display without going to previous step! see I2)

### English

1) [x] Online help (theory, instruction)
    - [ ] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
    - with default parameters ok, besides that same behavior as described in German
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Textual:**
- Online help:
    - Short introductory "Welcome to our online help for the SSL/TLS handshake plugin.". I'm not a fan of it, would remove it. It is however very short which could be acceptable. Should make general decision on this.
    - "Step 5/7", "Step 6/8" -- confusing, not explained
    - External online help links work here (on Linux/GTK3)! s. extra issue

### German

- Online Help:

1) [x] online help (theory, instruction)
    - [ ] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
    - cipher suite random selections and proceeding as in I1) results in erroneous behavior after step 1 in client. Unacceptable without some kind of message; graphics glitches (arrows).
3) [x] walk-through, different order
4) [x] Typos
    - some texts and labels are still in English, see I3)
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Textual:**
- Online help:
    - same as in English
    - "Schritt-für-Schritt" is "word-by-word" translation --> "Schritt für Schritt"
    - External online help links work here (on Linux/GTK3)! s. extra issue

**I1)**

![ssl_tls_hs1](https://user-images.githubusercontent.com/982379/77823404-39614680-70fb-11ea-8bfa-8936cd423544.gif)


**I2)**

![ssl_tls_hs2](https://user-images.githubusercontent.com/982379/77823402-36665600-70fb-11ea-8572-40f5bc4f8590.gif)


**I3)**

![ssl_tls_hs3](https://user-images.githubusercontent.com/982379/77823400-336b6580-70fb-11ea-927a-1a4effb719a9.png)

**Comment by simlei** [comment]
## Verifiable Secret Sharing

### Preliminary verdict:

- Near-perfect -- no major issues, looks nice, no glitches, good help.

### English

1) [x] Online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### German

1) [x] online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Comment by simlei** [comment]
## WOTS / WOTS+

### Preliminary verdict:

- Pretty good, up to the badly-scaled visual -- no major issues, looks nice, no glitches, good help.
- **Medium** prio: visual investigate scaling issue.

### Both languages

- Visual bottom left: bad scaling? s. I1) (Linux/GTK)

### English

1) [x] Online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [x] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### German

1) [x] online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [x] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**I1)**

![wots1](https://user-images.githubusercontent.com/982379/77824246-95c76480-7101-11ea-80e1-dcdc9c359317.png)

**Comment by simlei** [comment]
## Simple Power Analysis / Square and Multiply

### Preliminary verdict:

- **low/medium** UI is a bit rough on the edges and does not notify of insufficient input on subsequent iterations (I2). Nothing critical though.

### English

1) [x] Online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
2) [x] Walk-through major functions
    - Wenn man "Chiffretext" leer lässt, wird kein Fehler angezeigt, aber die Ausgabe bleibt unvollständig. Es sollte so nicht ausgeführt werden dürfen: I2)
3) [x] Walk-through, different order
    - Nachdem einmal durchlaufen wurde: wenn man nun versucht in "Chiffre*text*" einen Text einzugeben passiert nichts. Es sind nur Zahlen erlaubt. Der Mouseover-Text erklärt es, aber man muss ihn erst finden.
4) [x] Typos
    - Deutsche Texte sind teilweise nicht ideal: I1)
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Textual:**
- Online help:
    - _"Ciphertext c(e.g. 5454)"_ (missing space)
    - "_copime_" --> "coprime"
    - "_Shown in the figure..._" --> "As shown in the figure..."
    - "_sub section_" --> "subsection"
    - "_, or graphs of electrical activity over time_" --> ", or graphs of electrical activity, over time"

### German

1) [x] online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
2) [x] walk-through major functions
    - s. Englisch
3) [x] walk-through, different order
    - s. Englisch
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Textual:**
- Online help:
    - `<\p>` artifacts in online help (e.g. at the end). The problem is with the backslash, it should be a slash.

**I1)**

![squaremulti1](https://user-images.githubusercontent.com/982379/77825223-3e78c280-7108-11ea-945b-8e2167285979.gif)


**I2)**

after executing with insufficient input, only a part of the usual results appears (empty input -> no op?), but nothing notifies the user of the limited value to be had from that result.

![squaremultiply2](https://user-images.githubusercontent.com/982379/77825231-420c4980-7108-11ea-9570-a747595b8d05.png)

**Comment by tassadarius** [comment]
# [Game] Divider Game

:heavy_check_mark:  looks good
:yellow_heart:  may need some love
:red_circle: critical bug, should be fixed

**Update: https://github.com/jcryptool/crypto/pull/290**
**Tested Platforms:**
 * Windows 10
 * Linux (i3wm, gtk)

----

### UI :heavy_check_mark:

* [x] Resize artifacts  
* [x] Layout consistency

**Problems**
 * ~~Visual Feedback not optimal (current player could be displayed in the middle, winning player is shown not well, reset button could be more prominent).~~
 * ~~Computer moves could be shown with a delay so the human clearly sees what the computer picked.~~
 * ~~Reset button breaks disabling the strategy button in player mode~~
 * ~~MessageBoxes appear on wrong screen~~

## English

### Online help (only background, little instruction, quite short)  :heavy_check_mark::
 * [x] has context help
  * ~~Typo in online help "looses" :arrow_right: "loses"~~
 * [x] has online help
  * ~~Typo in online help "looses" :arrow_right: "loses"~~
  * ~~Only show english wikipedia link in english online help~~
  * ~~People may not know what NRW is, may be clearer~~
 * [x] has toolbar button that shows context help

### Plugin  :heavy_check_mark:
* [x] Walk-through major functions
* [x] Walk-through, different order
* [x] Typos
 
## German

### Online help :heavy_check_mark::

* [x] has context help
* [x] has online help
  * ~~Only show German wikipedia link in German online help~~
  * ~~People may not know what NRW is, may be clearer~~
* [x] has toolbar button that shows context help
* [x] do online help images scale with page width?

### Plugin  :heavy_check_mark:

* [x] walk-through major functions
* [x] walk-through, different order
* [x] Typos

### Textual Problems

* ~~Some button descriptions in the top right not German~~
* ~~Main description:   Im Spielfeld werden ~~dann nur noch~~ die Teiler der Anfangszahl angezeigt~~

**Comment by tassadarius** [comment]
# [Game] Number Shark

:heavy_check_mark:  looks good
:yellow_heart:  may need some love
:red_circle: critical bug, should be fixed

### UI :heavy_check_mark: :

* [x] Resize artifacts  
* [x] Layout consistency

**Problems**

 * Has unnecessary TabItem (only one tab)
 * Unnecessary view Menu entry in top right corner
  * Plugin window has no min size

## English

### Online help (very detailed)  :yellow_heart: :
 * [x] has context help
  * may explain a little how the game works
 * [x] has online help
  * Link to http://www.maa.org/MathHorizons is dead 
 * [x] has toolbar button that shows context help
* [ ] do online help images scale with page width? :yellow_heart:

### Plugin :red_circle: 
* [x] Walk-through major functions
* [x] Walk-through, different order
* [x] Typos

**Problems**
* Outputs of solution wizards disappear immediatly :red_circle: see Issue #266 
* Game does not self explain how it works, you have to contact online help

### Textual Problems
* "Heuristic strategies" wizard: "Size of the __studied__ playing fields"
* "Optimal strategies" wizard: "Size of the __studied__ playing fields"


## German

### Online help (very good and detailed online help)  :yellow_heart::
* [x] has context help
 * may explain a little how the game works
* [x] has online help
 * Site "Diskussion" and "Anhang" only available in English 
 * Link to http://www.maa.org/MathHorizons is dead 
* [x] has toolbar button that shows context help
* [ ] do online help images scale with page width? :yellow_heart:

### Plugin  :red_circle: :

* [x] walk-through major functions
* [x] walk-through, different order
* [x] Typos

**Problems**
* Game does not self explain how it works, you have to contact online help
* Outputs of solution wizards disappear immediatly :red_circle: see Issue #266
### Textual Problems

* Some button descriptions in the top right not German, e. g. Save or  Minimize (may be bigger problem than just Plugin)
* Some yes/no buttons are in English
* Title in Fenster --> Spiele is in English
* Main description:   Im Spielfeld werden ~~dann nur noch~~ die Teiler der Anfangszahl angezeigt

**Comment by simlei** [comment]
@tassadarius that three-colour system is nice, I think I'll remaster my reviews to use these some time the next days :)

**Comment by simlei** [comment]
@ **[Game] Number Shark**

> * Link to http://www.maa.org/MathHorizons is dead

Does not look dead to me

**Comment by Leikani** [comment]
I have a look at it. The link works, but within JCT it doesn't.

Am 28.03.2020 um 23:47 schrieb Simon Leischnig:
>> * Link to http://www.maa.org/MathHorizons is dead
> Does not look dead to me
>
>

**Comment by simlei** [comment]
> I have a look at it. The link works, but within JCT it doesn't.

I'm already tracking that:
https://github.com/jcryptool/crypto/issues/289#issuecomment-605257525

It seems there is a way to do external links that at least under linux they work.
It also seems that there is a way to to this that they don't. Not necessarily the same under every OS.

Best place to discuss systematic issues with the plug-ins is there (where it will get sorted into single issues in time) or in the existing issues.

**Comment by tassadarius** [comment]
# [Game] Sudoku Solver

:heavy_check_mark:  looks good
:yellow_heart:  may need some love
:red_circle: critical bug, should be fixed

### UI :yellow_heart: to :heavy_check_mark: :

* [x] Resize artifacts  
* [x] Layout consistency

**Problems**

 * Hexadecimal Sudoku larger than a height of 1080p. So with a "default" HD monitor you always have to scroll (may be actually not possible to fit to 1080p)

## English

### Online help  :yellow_heart::
 * [ ] has context help :yellow_heart: 
 * [ ] has toolbar button that shows context help :yellow_heart: 
 * [x] has online help

**Problems**

 * `Missing`: Does not have any pictures / images :yellow_heart:
 * `Bad Wording`: __Sudoku Tactics__ --> "A killer sudoku is much harder to brute force because of the stranger restrictions"
 * `Dead Link`: www.su-doku.net --> http://www.su-doku.net/tech.php"
* `Wrong Link`: www.sudokuoftheday.com actually links to  http://localhost/help/topic/org.jcryptool.games.sudoku/$nl$/help/content/ and is then not displayed
* `Misleading Wording`: __Plug-in Instructions__ --> Once all the squares in an area are selected, choose the operator from the menu on the left and enter therequired result.
* `Misleading Wording`: __File Handling__ --> A user may not know what a \n is

### Plugin :red_circle: to :yellow_heart: : 
* [x] Walk-through major functions
* [x] Walk-through, different order
* [x] Typos

**Problems**
* `Wording`: I would call the *"Enter"* button actually *"Create"*. You can enter many things, that is not clear
* `Missing User Feedback`: In the Killer Sudoku a divison may only be of 2 squares. If you select more it just disappears without info why. Seems like a bug the.
* `Uncovered Error Handling`: The solver sometimes runs into an StackOverflowError, especially in the Hexadecimal world and then displays the error. It should be catched and presented to the User that it was not solvable. :red_circle: 
* `General`: The *"Autofill"* button has a wrong description and is misleading in my opinion. It does the same as Hint but only more specifically with minor benefit to the game.
* `General`: The possible solutions in the Killer Sudoku seem arbitrary as not every box has a suggestion without any obvious reason.
* `General`: There's no indicator when you correctly solved the sudoku. It is just filled out. Leaves you a little bit disappointed.
* `General (Performance)`: The solving algorithms for Killer and Hexadecimal take a long time. This may be inconvenient for the user, as it is not obvious how to stop the calculation.
* `Coloring`: Red/Green coloring of the Button *"Show Possible"* looks horrible and is not color blind friendly. 
* `Suggestion`: The reset button could become a reset button when creating a Sudoku

## German

### Online help  :yellow_heart::

 * [ ] has context help :yellow_heart: 
 * [ ] has toolbar button that shows context help :yellow_heart: 
 * [x] has online help

**Problems**

* `Spelling`: **Sudoku-Löser** --> hexadezimale**s** Sudokus
* `Wording`: **Sudoku-Lösungstaktik** --> Trotzdem sind auch dazu momentan keine Taktiken in das Plug-in eingebaut, weil wir keine "guten" Taktiken fanden".
* `Inconsistency`: **Sudoku-Lösungstaktik** --> Paragraph 1 and 3 contain contradictory information.
* `Formatting`: **Datei-Support** --> paragraph has different font.
* `Spelling`: **Beispiel für eine Killer-Sudoku-Datei** --> Einkomplettes
* `Spelling`: **Erklärung der Struktur der Killer-Sudoku-Datei** --> Killer Sudoku (missing -)

### Plugin :red_circle: to :yellow_heart: : 

* [x] walk-through major functions
* [x] walk-through, different order
* [x] Typos

**Problems**

* `Wording`: Description of *"Standard-Puzzle"* says "Zeigt ein zufälliges Standard-Puzzle". Should rather say "Lädt ein zufällig gewähltes Standard-Puzzle".
* `Wording`: Button *"Löse evt. nächste"* is confusing and has a different tooltip than the English button.
* `Wording`: Button *"Speichere Puzzle"* tooltip "Sichert das aktuell angezeigte Puzzle in eine *.SUD -Datei"--> "Speichert das aktuelle Puzzle in eine .sud-Datei"


For some problems see the English problems also

**Comment by simlei** [comment]
## Huffman Coding 

- Tested under Linux/GTK3
- Tested under Win by @grthor 

### Preliminary verdict:

- Positive example on how to use ZEST framework under linux! graph in second tab is A-OK!
- No UTF-8, Malfunction if used for it (**high**)
- Context help

### Checklist English

1) [x] Online help (theory, instruction)
    - [ ] has context help *(but only on F1!)*
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### Checklist German

1) [x] online help (theory, instruction)
    - [ ] has context help *(but only on F1!)*
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

Notes

Labels/Textfields/Buttons:

- ~~Titel-Label hat zu viele Zeilen s. Screenshots~~ fixed with https://github.com/jcryptool/crypto/commit/df771f8a5c16f971710424f3c3a0e8a08331d45c by @grthor 
- ~~cut off labels in English (see first screenshot)~~ fixed with https://github.com/jcryptool/crypto/commit/df771f8a5c16f971710424f3c3a0e8a08331d45c by @grthor 

Online Help

- "Textdateien (ISO-8859-1) können geladen werden" -- nach Umstellen auf UTF-8 ändern bzw. ganz weglassen

Malfunction:

- huffman1.png: with funny characters or just an "Ü": 0 bytes output, "100% kompressionsrate". Text see below.
- Huffman2.png: when loading the same text from file (utf-8 encoded), the characters are funny and the compression rate is negative.
- "Pfad anzeigen" Button in 3rd tab: switches to the second tab alright but toolbar controls are not shown (zoom etc.)
- same in German and English

Text:
```
Dies ist ein kurzer Beispieltext
in different Languages
with fûnæ ©ħa¶æ¢ŧ€®ẞ
```

![Cut-off labels](https://user-images.githubusercontent.com/982379/78797471-07cc6300-79b8-11ea-8307-ed6019ea35ab.png)
![huffman1](https://user-images.githubusercontent.com/982379/78797076-81b01c80-79b7-11ea-87f3-1afeb65698c4.png)
![Huffman2](https://user-images.githubusercontent.com/982379/78797075-81178600-79b7-11ea-9d96-ef42e03ce598.png)

**Comment by simlei** [comment]
## Inner States of DES

- Tested under Linux/GTK3
- Tested under Win by @grthor 

### Preliminary verdict:

- ~Context help is wrong; @grthor :The content help contains of "Inner States of DES". This is not enough.~ (fixed)
- ~online help is just the link to the paper which is not enough in 1 case at least~ (fixed with additional link in the error message)

### Checklist English

1) [x] Online help (theory, instruction)
    - [ ] has context help (context help on F1 goes to homomorphic enc.)
    - [ ] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### Checklist German

1) [x] online help (theory, instruction)
    - [ ] has context help (context help on F1 goes to homomorphic enc.)
    - [ ] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

Notes

Online Help

- No help as to how to use the GUI
    - see screenshot 1 -- followed GUI instructions but "Parität .. ist falsch". Do I have to read the paper now? (same in EN)

![isdes](https://user-images.githubusercontent.com/982379/78802569-45cc8580-79be-11ea-8049-c3c89d61c927.png)

**Comment by simlei** [comment]
## Kleptography 

- Tested under Linux/GTK3
- Tested under Win 10 by @grthor 

### Preliminary verdict:

- Funny/to fix: Context Help/pressing F1: works only when having clicked on certain parts of the GUI (e.g. explanation of steps text fields) -- @grthor : In a superficial analysis I identified that F1 works if the user clicked on a GUI part that uses grid layout, if the user clicked on a part which uses stack layout it did not work.
- Rest is pretty good!

### Checklist English

1) [x] Online help (theory, instruction)
    - [x] has context help (does not react to view focus though)
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### Checklist German

1) [x] online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Comment by simlei** [comment]
## Merkele Signature  

- Tested under Linux/GTK3
- Tested under Win10 by @grthor 

### Preliminary verdict:

- WOW to the robustness and functioning of the visualization trees!! Not a single issue found over all paths through the plugins!
- ~~The plugin could need an overarching title-and-description area, not "just" in the tabs. (minor nitpick)~~ adressed with https://github.com/jcryptool/crypto/commit/e4a42efb7f35478c58a56d416500cb258e6cc18cby by @grthor 

### Checklist English

1) [x] Online help (theory, instruction)
    - [x] has context help fixed by @grthor 
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### Checklist German

1) [x] online help (theory, instruction)
    - [x] has context help fixed by @grthor 
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Comment by simlei** [comment]
## Merkle-Helman Knapsack Cryptosystem 

- Tested under Linux/GTK3
- Tested under Win10 by @grthor 

### Preliminary verdict:

- ~Context help missing~ fixed by @grthor with https://github.com/jcryptool/crypto/commit/c1fbc1a5c14ef8ad32a0dd8039720ef20682bf28
- else all pretty normal/good

### Checklist English

1) [x] Online help (theory, instruction)
    - [x] has context help fixed by @grthor with https://github.com/jcryptool/crypto/commit/c1fbc1a5c14ef8ad32a0dd8039720ef20682bf28
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### Checklist German

1) [x] online help (theory, instruction)
    - [x] has context helpfixed by @grthor with https://github.com/jcryptool/crypto/commit/c1fbc1a5c14ef8ad32a0dd8039720ef20682bf28
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [ ] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Comment by simlei** [comment]
## Grille

- Tested under Linux/GTK3
- Tested under Win 10 (@grthor )

### Preliminary verdict:

- **Does not encrypt UTF-8 correctly**
- ~~Resize problem~~ Fixed with https://github.com/jcryptool/crypto/commit/e637c6cf17611f82382845eef2d01b1352cbcdca @grthor 
- visualization (Grille square) shows standard 7-bit chars sometimes funny, too (screenshot 1). Presumably, those are line breaks -- should be replaced with `\n`; @grthor This are not only linebreaks. This are normale spaces too. I am also in favour of line breaks being represented as `\n`

### Checklist English

1) [x] Online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [x] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
    - ~~on resizing, the lower part may be cut off (see gif)~~ Fixed with https://github.com/jcryptool/crypto/commit/e637c6cf17611f82382845eef2d01b1352cbcdca @grthor 
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### Checklist German

1) [x] online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [x] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
    - ~~on resizing, the lower part may be cut off (see gif)~~ Fixed with https://github.com/jcryptool/crypto/commit/e637c6cf17611f82382845eef2d01b1352cbcdca @grthor 
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

Notes

Malfunction:

 - UTF-8 characters are not processed correctly, see screenshot 1. The text used is as follows:
 - @grthor : UTF8 chars work on windows, see the thrid screenshot.
 - ~~The encryption/decryption does not work correctly. If you use the default text from the text editor and encrypt it, then decrypt it the text is not the same. I guess the grille is turned one time to much or too less.~~ @grthor Fixed with https://github.com/jcryptool/crypto/commit/0aa546ff0af36caaf7b434317192f092ef31e08f

```
Dies ist ein kurzer Beispieltext
in different Languages
with fûnæ ©ħa¶æ¢ŧ€®ẞ
```
![Grille_resize](https://user-images.githubusercontent.com/982379/78911908-454bf180-7a87-11ea-9fea-aa803febc8cc.gif)

![grille_1](https://user-images.githubusercontent.com/982379/78911913-467d1e80-7a87-11ea-8e31-eb430b5c8b1f.png)

![image](https://user-images.githubusercontent.com/20046726/79317415-fb8b5d00-7f05-11ea-8830-4230b3997026.png)

**Comment by simlei** [comment]
## Hash Sensitivity

- tested on Linux by @simlei 
- tested on Windows 10 by @grthor 

### Preliminary verdict:

- ~~No scrolled composite as root of view~~ fixed with https://github.com/jcryptool/crypto/commit/35ff721fc8dc39e1b74ef116bc5f9ff88d48f8d7 by @grthor 
- ~~No context help~~ fixed with https://github.com/jcryptool/crypto/commit/d7f362330dfbb81d4197d9e77c3e58b156c1d5ae by @grthor 
- ~~missing image in online help (EN)~~ fixed with https://github.com/jcryptool/crypto/commit/e982db526fc32cf329cbb826515428438bf6463d by @grthor 
- Else, pretty solid!

### Checklist English

1) [x] Online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### Checklist German

1) [x] online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

**Comment by simlei** [comment]
## RSA Cryptosystem 

- Tested under Linux/GTK3


### Preliminary verdict:

- Pretty instable and prone to various errors (see below in Notes)

--> Thorough test and bugfix warranted.

- Help is OK I think but does not contain any images.

### Checklist English

1) [x] Online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### Checklist German

1) [x] online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

Notes

Malfunction:

- **Catastrophic failure** under some condition (Alice's key) in all four tabs: Modulus N is negative INT_MAX, affects everything else. (see first screenshots)
    - probably due to Alice's key not being valid? Should be properly handled, though!
    - entering input as number blocks fails: negative mod (seems like (-1) * INT_MAX ) - screenshot 1
    - after clicking on "eigenes Alphabet erstellen" (screenshot 2), "phantom" shell appears and **cannot be closed**!! (maybe linux specific). Also, the wrong modulus is still in effect.
    - Rest of the plug-in does not work correctly: Decryption does not yield the right plain text, input of steps into dialogs doesnt work right (modulus error), etc
- When in a state of "editor disarray" (view maximized, editor crashed/not present) (second gif): No possibility to enter text, not even manually (dialog does not appear): not able to proceed to test en/decryption. Would be fixed by fixing the editor/maximizing problem **but the dialog should catch this error** even if it is not yet fixed.
- Encryption: calculation hiccups as shown in the second gif -- next / previous step is disabled
- **Ghost shells** in dialogs that cannot be closed until the dialog closes (under linux at least)

![rsa1](https://user-images.githubusercontent.com/982379/79195995-864d5880-7e2f-11ea-81ca-1110baa3c8ce.gif)
--------

![rsa2](https://user-images.githubusercontent.com/982379/79195990-84839500-7e2f-11ea-9922-ee0b0d6cf19a.gif)

----------

![rsa2](https://user-images.githubusercontent.com/982379/79195996-86e5ef00-7e2f-11ea-939b-a1be64fec4ce.png)

--------

![rsa1](https://user-images.githubusercontent.com/982379/79195997-877e8580-7e2f-11ea-88ae-3774fbc56e80.png)

**Comment by simlei** [comment]
## Magic Door  

- Tested under Linux/GTK3
- Tested under Win 10 by @grthor 

### Preliminary verdict:

- **Partial disfunct** dialog irresponsive even after resetting via toolbar button; (@grthor if you mean the "Run several times"-button this is fixed https://github.com/jcryptool/crypto/commit/879eb58a5e1c5fcd3c52c59499992d4064fae525.
### Checklist English

1) [x] Online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### Checklist German

1) [x] online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

Notes

Labels/Textfields/Buttons:

- ~~"Run several times" dialog may be perceived to be frozen: clicking on radio button texts does not toggle them~~ @grthor : Fixed with https://github.com/jcryptool/crypto/commit/879eb58a5e1c5fcd3c52c59499992d4064fae525; Clicking on the text sets the corresponding radio button,

![magicdoor1](https://user-images.githubusercontent.com/982379/79196305-22775f80-7e30-11ea-82e8-158824e4ac46.png)

**Comment by simlei** [comment]
## Homomorphic Encryption
- Tested under Linux/GTK3
- Tested under Win 10 by @grthor 
### Preliminary verdict:

- Bug bei im Nachhinein angeketteter Operation bei Gentry & Hallevi (s. Notes)?

### Checklist English

1) [x] Online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] Walk-through major functions
3) [x] Walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

### Checklist German

1) [x] online help (theory, instruction)
    - [x] has context help
    - [x] has online help
    - [ ] has toolbar button that shows context help
    - [x] do online help images scale with page width?
2) [x] walk-through major functions
3) [x] walk-through, different order
4) [x] Typos
5) [x] Resize artifacts
6) [x] Layout consistency: e.g. Buttons and groupings vertical, horizontal alignment
7) [x] Dialogs: autofocus on first field?

## Notes

Labels/Textfields/Buttons:

- Tab Gentry & Halevi: in "Entsprechende Operationen auf dem Klartext (Group) -> Operationen (Textfeld)": hier sollte noch "mod n" hinter dem Term stehen.

Malfunction:

- (Mit angelegtem Schlüssel im Keystore) Tab RSA -> Ergebnis Entschlüsseln (Schlüsselauswahldialog) -> mit erstem voreingestellten existierendem Schlüsselpaar ist keine "Fertigstellen" möglich. Ich muss da erst einen anderen Schlüssel auswählen und dann den gewünschten.
- In Tab Gentry und Halevi: Nach erfolgreichem ver- und Ergebnis ent-schlüsseln von n=128, o1=33, o2=55, op=Addieren (Ergebnis: 88) addierte ich danach noch mal 2 hinzu und klickte "Ergebnis entschlüsseln". Nun kommt Ergebnis = 90, aber Entschlüsselt (Dezimalzahl) = 26. Das ist ein Bug, wenn mich nicht alles täuscht. S. Screenshot.
    - bei Tabs RSA und Paillier scheint das aber zu funktionieren
    - Die Hilfe erklärt mir: A somewhat homomorphic encryption scheme allows both homomorphic operations to be performed on the ciphertext, but the amount of consecutive operations is limited.
        - Doch Gentry & Hallevi soll ja "fully homomorphic" sein -- ??

![homomorph1](https://user-images.githubusercontent.com/982379/79198873-6cfadb00-7e34-11ea-8963-dd60cc4e713d.png)
![homomorphic2](https://user-images.githubusercontent.com/982379/79198876-6e2c0800-7e34-11ea-88ee-dcfdb56fba1a.png)

**Comment by simlei** [comment]
I have left this plugin in the milestone tracker for v1.0. However, I would like to close it soon! It served well as a testing organization place, but it is not ideal to separate out assignments of people to single tasks and tracking progress. 

I tried to move all unfinished work from this issue into their own issues. If something is missing, please add it to the resp. issue.

**Comment by grthor** [comment]
Would agree on closing this issue. 

It has served its purpose, but has now become too big.

It is much better to handle the remaining problems in individual issues.
